docs/tutorials/constraints.rst

```diff
@@ -0,0 +1,238 @@
+        Vega            True              True
+     Albireo            True             False
+       Algol            True             False
+       Rigel           False             False
+     Regulus           False             False
+
+Now we can see which targets are observable!
+
+.. _constraints-user_defined_constraints:
+
+User-Defined Constraints
+========================
+
+There are many possible constraints that you could find useful which have not
+been implemented (yet) in astroplan. This example will walk you through creating
+your own constraint which will be compatible with the tools in the `constraints`
+module.
+
+We will begin by reading the text file of stellar coordinates defined in the
+example above, and define an observer at Subaru::
+
+    from astroplan import Observer, FixedTarget
+    from astropy.time import Time
+    subaru = Observer.at_site("Subaru")
+    time_range = Time(["2015-08-01 06:00", "2015-08-01 12:00"])
+
+    # Read in the table of targets
+    from astropy.io import ascii
+    target_table = ascii.read('targets.txt')
+
+    # Create astroplan.FixedTarget objects for each one in the table
+    from astropy.coordinates import SkyCoord
+    import astropy.units as u
+    targets = [FixedTarget(coord=SkyCoord(ra=ra*u.deg, dec=dec*u.deg), name=name)
+               for name, ra, dec in target_table]
+
+In the above example, you may have noticed that constraints are assembled by
+making a list of calls to the initializers for classes like `AltitudeConstraint`
+and `AirmassConstraint`. Each of those constraint classes is subclassed from
+the abstract `Constraint` class, and the custom constraint that we're going to
+write must be as well.
+
+In this example, let's design our constraint to ensure that all targets must be
+within some number of degrees from Vega – we'll call it
+`VegaSeparationConstraint`. Two methods, `__init__` and `compute_constraint`
+must be written for our constraint to work:
+
+* The `__init__` method will accept the minimum and maximum acceptable separations
+  a target could have from Vega.
+
+* We'll also define a method `compute_constraints` which takes three arguments:
+  an array of times to test, an `Observer` object, and one or a list of
+  `FixedTarget` objects. `compute_constraints` will return a matrix of
+  booleans that describe whether or not each target meets the constraints.
+  The super class `Constraint` has a `__call__` method which will run your
+  custom class's `compute_constraints` method when you check if a target is
+  observable using `is_observable` or `is_always_observable`.
+
+Here's our `VegaSeparationConstraint` implementation::
+
+    from astroplan import Constraint, is_observable
+    from astropy.coordinates import Angle
+
+    class VegaSeparationConstraint(Constraint):
+        """
+        Constraint the separation from Vega
+        """
+        def __init__(self, min=None, max=None):
+            """
```

Comment:
I'd suggest to put the `__init__` docstring in the class-level docstring, in a `Parameters` section.

Either way will work, but having the `__init__` parameters documented in the class-level docstring is how I think most (all?) classes in Astropy are documented.